### ISSUE TITLE
Google login for Compainion

### ISSUE THREAD

**Issue by GOTFire** [mit-cml/appinventor-sources#1853]
Is it possible add Google login to companion? Its make faster app development.We need just login google account same as ai2 account.So we dont need busy with reading qrcodes.

**Comment by ewpatton** [comment]
It is something we are considering, but it hasn't really been part of our typical usage scenario. The current connection mechanism is aimed for schools with shared devices where there isn't a 1-1 correspondence between student and device. The issue with logging in is that you need to remember to log out, otherwise the next student will have access to all of your projects. The QR code mechanism, by design, allows you to quickly exit the app and then reenter with a new student without the risk of leaking information across sessions.

**Comment by ewpatton** [comment]
I'll also note that this had additional complications for our other servers that don't use Google login. code.appinventor allows for anonymous login, the Ghangzhou server uses QQ, and our HK instance is wrapped in an educational portal infrastructure that uses its own (non-Google) accounts.

**Comment by pavi2410** [comment]
Also, using QR codes/key input is better way to connect with emulators than a login system.

**Comment by GOTFire** [comment]
Yes , qrcode is better for students and emulators but i think too many people use their devices for debug ,like me( we need better emulator :(  ).

I think we need qrcode and google login at the same time.So users can select own method.I can select Google Login  and the students or emulator users can select qrcode login.

Google login method is not avaible for all user as ewpatton said.Maybe we can create somethink like api key.So we just enter this key for one time on companion .With this key we dont need any login provider.We can get this key on App Inventor 2 website ... 

improvement for api key method,
Maybe we can add  dropdown option for "Connect" menu.Like Connect > Add device for companion

So when user click this option we can show one time qrcode and its pass keys to ai2 companion.With this method we dont need enter key manually to ai2 companion.
